satpy/readers/seviri_l1b_icare.py

```diff
@@ -254,13 +254,13 @@ def get_dataset(self, ds_id, ds_info):
         offset = data.attrs.get('add_offset')
         scale_factor = data.attrs.get('scale_factor')
         data = data.where(data != fill)
-        data.values = data.values.astype(np.float32)
+        data.data = data.data.astype(np.float32)
```

Comment:
I don't understand why you need to apply this to `data.data` instead of just `data`? `xr.DataArray` supports this operation in principle.

Reply:
True, fixed.